src/serialize.ts

```diff
@@ -0,0 +1,61 @@
+import { Iterable, Record } from 'immutable';
+import JSONStreamStringify from 'json-stream-stringify';
+import {
+  SerializationOptions,
+  SerializationStreamOptions,
+} from './types/options';
+import {
+  isSupportedNativeType,
+  patchNativeTypeMethods,
+} from './utils/nativeTypes';
+import { serializeData, serializeDataAsync } from './utils/serializers';
+
+export function serialize(
+  data: any,
+  options: SerializationOptions = {},
+): string | never {
+  if (
+    Iterable.isIterable(data) ||
+    data instanceof Record ||
```

Comment:
There is a new function now that replaces `Iterable.isIterable` to check if something is Immutable: https://immutable-js.github.io/immutable-js/docs/#/isImmutable -- this will also cover `Record`.

... unless you're trying to support both Immutable 3 and 4? If so, then personally I'd suggest either something like:

```js
const immutableCheckFn = typeof Immutable.isImmutable === 'function' ? Immutable.isImmutable : Iterable.isIterable;
if (immutableCheckFn(data)) { //...
```

or duck typing:

```js
if (typeof data.toJS === 'function') { //...
```

Reply:
@idolize This doesn't update immutablet yet

Reply:
Ah, in that case the `data instanceof Record` check should never trigger since `Iterable.isIterable` returns `true` for Records as well.